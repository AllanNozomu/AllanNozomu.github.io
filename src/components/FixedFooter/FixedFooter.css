.heart {
    color: #EB5252;
}

.App-footer a{
    color: black;
}

.last-update {
    display: none;
}

@media screen and (min-width: 768px){
    .last-update {
        display: initial;
    }
}