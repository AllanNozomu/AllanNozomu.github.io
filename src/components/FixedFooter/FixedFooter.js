import React from 'react'
import './FixedFooter.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const FixedFooter = (props) => {
  const {lastTimeUpdate} = props

  return (
    <footer className="App-footer">
      <p>
        {`Created with `}
        <FontAwesomeIcon icon="heart" className="heart animated-char" /> 
        {` by `} 
        <a href="https://github.com/allannozomu/my-site" target="#">allannozomu</a>
        <span className="last-update"> {` | Last update on ${lastTimeUpdate}`} </span>
      </p>
    </footer>
  )
}

export default FixedFooter