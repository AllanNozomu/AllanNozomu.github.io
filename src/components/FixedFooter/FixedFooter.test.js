import React from 'react'
import { shallow } from 'enzyme';

import FixedFooter from './FixedFooter';

describe('FixedFooter', () => {
  it('Footer without last update date', () => {
    const component = shallow(<FixedFooter />);

    expect(component).toMatchSnapshot();
  });
  
  it('Footer with last update date', () => {
    const component = shallow(<FixedFooter lastTimeUpdate="11/12/2013"/>);

    expect(component).toMatchSnapshot();
  });
})