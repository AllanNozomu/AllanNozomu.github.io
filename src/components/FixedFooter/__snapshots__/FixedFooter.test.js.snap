// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedFooter Footer with last update date 1`] = `
<footer
  className="App-footer"
>
  <p>
    Created with 
    <FontAwesomeIcon
      border={false}
      className="heart animated-char"
      fixedWidth={false}
      flip={null}
      icon="heart"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     by 
    <a
      href="https://github.com/allannozomu/my-site"
      target="#"
    >
      allannozomu
    </a>
    <span
      className="last-update"
    >
       
       | Last update on 11/12/2013
       
    </span>
  </p>
</footer>
`;

exports[`FixedFooter Footer without last update date 1`] = `
<footer
  className="App-footer"
>
  <p>
    Created with 
    <FontAwesomeIcon
      border={false}
      className="heart animated-char"
      fixedWidth={false}
      flip={null}
      icon="heart"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     by 
    <a
      href="https://github.com/allannozomu/my-site"
      target="#"
    >
      allannozomu
    </a>
  </p>
</footer>
`;
